api/applications/applications_handler.go

```diff
@@ -458,6 +456,34 @@ func (ah *ApplicationHandler) TriggerPipelinePromote(ctx context.Context, appNam
 	return jobSummary, nil
 }
 
+func (ah *ApplicationHandler) getRadixDeployment(ctx context.Context, appName string, envName string, deploymentName string) (*v1.RadixDeployment, error) {
+	if len(deploymentName) < 5 || len(deploymentName) > 200 {
+		return nil, errors.New("invalid deployment name")
+	}
+	radixDeployment, err := kubequery.GetRadixDeploymentByName(ctx, ah.accounts.UserAccount.RadixClient, appName, envName, deploymentName)
+	if err == nil {
+		return radixDeployment, nil
+	}
+	if !k8serrors.IsNotFound(err) {
+		return nil, fmt.Errorf("failed to get deployment %s for the app %s, environment %s: %v", deploymentName, appName, envName, err)
+	}
+	envRadixDeployments, err := kubequery.GetRadixDeploymentsForEnvironment(ctx, ah.accounts.UserAccount.RadixClient, appName, envName)
+	if err != nil {
+		return nil, err
+	}
+	radixDeployments := slice.Reduce(envRadixDeployments, []*v1.RadixDeployment{}, func(acc []*v1.RadixDeployment, rd v1.RadixDeployment) []*v1.RadixDeployment {
+		radixDeploymentSuffix := fmt.Sprintf("-%s", deploymentName)
+		if strings.HasSuffix(rd.Name, radixDeploymentSuffix) {
+			acc = append(acc, &rd)
+		}
+		return acc
+	})
+	if len(radixDeployments) != 1 {
+		return nil, errors.New("invalid or not existing deployment name")
+	}
+	return radixDeployments[0], nil
```

Comment:
```suggestion
    radixDeploymentSuffix := fmt.Sprintf("-%s", deploymentName)
	radixDeployments := slice.FindAll(envRadixDeployments, func(rd v1.RadixDeployment) bool {return strings.HasSuffix(rd.Name, radixDeploymentSuffix)})
	if len(radixDeployments) != 1 {
		return nil, errors.New("invalid or not existing deployment name")
	}
	return &radixDeployments[0], nil
```
use slice.FindAll instead